### ISSUE TITLE
Update OneNote page content not working - incorrect url schema being used

### ISSUE THREAD

**Issue by parkerschroeder** [microsoftgraph/msgraph-sdk-java#1361]
### TL;DR
Updating existing OneNote pages does not work using the Graph Java SDK. It seems like an old url schema is being used that is no longer supported by OneNote/Graph. The new API can be found [here](https://learn.microsoft.com/graph/onenote-update-page).

### Expected behavior

According to the Graph API documentation for [updating OneNote page content](https://learn.microsoft.com/graph/onenote-update-page#request-with-text-content-only), the proper HTTP call to Update an existing OneNote page is 
```http
PATCH https://graph.microsoft.com/v1.0/me/onenote/notebooks/pages/{page-id}/content
```

In GraphSDK syntax, this would likely be expected to be something like `graphClient!!.me().onenote().pages(pageId).content.buildRequest().patch(byteArray)`

Using the Graph Explorer to confirm this, using a valid PATCH request with “/content” as the ending segment
```http
PATCH https://graph.microsoft.com/v1.0/me/onenote/pages/{page-id}/content
```
and the following JSON body
```json
[
  {
    "target": "title",
    "action": "replace",
    "content": "New title"
  }
]
```
returns a HTTP 204 (success)

The GraphSDK recommends a different approach, however, based on their [testPatchContent() test](https://github.com/microsoftgraph/msgraph-sdk-java/blob/1e9363e462b5446a432897b9b895d385bafd3ee8/src/test/java/com/microsoft/graph/functional/OneNoteTests.java#L521).
The test uses the code `graphClient!!.me().onenote().pages(pageId).onenotePatchContent(parameterSet).buildRequest().post()` to update an existing OneNotePage's contents.

The expected behavior is that this GraphSDK approach returns the same HTTP 204 (success) code as the PATCH approach above, but an error is thrown instead.

### Actual behavior
When calling the code `graphClient!!.me().onenote().pages(pageId).onenotePatchContent(parameterSet).buildRequest().post()` from the GraphSDK, the following error message is produced.
```
java.util.concurrent.CompletionException: com.microsoft.graph.http.GraphFatalServiceException: Unexpected exception returned from the service.Error code: 19999
Error message: Something failed, the API cannot share any more information at the time of the request.

POST https://graph.microsoft.com/v1.0/me/onenote/pages/{page-id-removed-here}/microsoft.graph.onenotePatchContent
SdkVersion : graph-java/v5.47.0 
500 : Internal Server Error

java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:290)
java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:295)
java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:621)
java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:482)
java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2049)
com.microsoft.graph.http.CoreHttpCallbackFutureWrapper.onResponse(CoreHttpCallbackFutureWrapper.java:34)
okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
java.lang.Thread.run(Thread.java:1012)

```

This can be confirmed by running the POST request in the Graph Explorer with a valid page id ([link to query here](https://developer.microsoft.com/graph/graph-explorer?request=me%2Fonenote%2Fpages%2F%7Bpage-id%7D%2Fmicrosoft.graph.onenotePatchContent&method=POST&version=v1.0&GraphUrl=https://graph.microsoft.com&requestBody=IntcImNvbW1hbmRzXCI6W3tcInRhcmdldFwiOlwidGl0bGVcIixcImFjdGlvblwiOlwicmVwbGFjZVwiLFwiY29udGVudFwiOlwiTmV3IHRpdGxlXCJ9XX0i&headers=W3sibmFtZSI6IkNvbnRlbnQtdHlwZSIsInZhbHVlIjoiYXBwbGljYXRpb24vanNvbiJ9XQ==))
```http
POST https://graph.microsoft.com/v1.0/me/pages/{page-id}/microsoft.graph.onenotePatchContent
```
with the modified JSON.
```json
{
    "commands": [
        {
            "target": "title",
            "action": "replace",
            "content": "New title"
        }
    ]
}
```
The same error seen using the GraphSDK is returned here as well, in the Graph Explorer
![image](https://user-images.githubusercontent.com/21373837/222562311-d31ecde9-f153-42bd-93ed-4142d2bc10eb.png)

### Steps to reproduce the behavior
 1. In Android Studio Dolphin 2021.3.1, use the following library versions for authentication and graph calls:
```
    implementation 'com.microsoft.graph:microsoft-graph:5.44.0'
    implementation 'com.google.guava:guava:30.1.1-android'
    implementation 'com.azure:azure-identity:1.2.5'
    implementation "com.microsoft.identity.client:msal:4.1.3"
```
2. Authenticate user with MSAL, then create graph client with resulting access token:
```
// inside onSuccess method of AuthenticationCallback
val authProvider = IAuthenticationProvider {
    CompletableFuture.supplyAsync { authenticationResult.accessToken }
}

scope.launch(Dispatchers.IO) {
    val graphClient = GraphServiceClient.builder().authenticationProvider(authProvider).buildClient()

    try {
        val commands: MutableList<OnenotePatchContentCommand> = mutableListOf()
        val command = OnenotePatchContentCommand()
        command.target = "body"
        command.action = OnenotePatchActionType.APPEND
        command.position = OnenotePatchInsertPosition.AFTER
        command.content =
            "<img src=\"https://en.wikipedia.org/wiki/File:Alexander_Hamilton_portrait_by_John_Trumbull_1806.jpg\" alt=\"New image from a URL\" />"
        commands.add(command)

        val contentRequest = OnenotePageOnenotePatchContentParameterSet.newBuilder().withCommands(commands)
            .build()

        graphClient!!.me().onenote()
            .pages(pageId)
            .onenotePatchContent(
                contentRequest
            )
            .buildRequest()
            .post()
    } catch (e: Exception) {
        Log.d(TAG, "${e.message}")
    }
}
```

OR

1. Sign into the [graph explorer](https://developer.microsoft.com/graph/graph-explorer)
2. Use the following query to configure the request settings - [query](https://developer.microsoft.com/graph/graph-explorer?request=me%2Fonenote%2Fpages%2F%7Bpage-id%7D%2Fmicrosoft.graph.onenotePatchContent&method=POST&version=v1.0&GraphUrl=https://graph.microsoft.com&requestBody=IntcImNvbW1hbmRzXCI6W3tcInRhcmdldFwiOlwidGl0bGVcIixcImFjdGlvblwiOlwicmVwbGFjZVwiLFwiY29udGVudFwiOlwiTmV3IHRpdGxlXCJ9XX0i&headers=W3sibmFtZSI6IkNvbnRlbnQtdHlwZSIsInZhbHVlIjoiYXBwbGljYXRpb24vanNvbiJ9XQ==)
3. Replace the `{page-id}` in the url with a valid OneNote page id associated with your account (can find one by running a "My Pages" request from the Graph Explorer)
4. Run the query and see the output for any error messages

**Comment by ramsessanchez** [comment]
Hi @parkerschroeder, it seems you have uncovered a big issue regarding the OneNote api. For context, we rely on the workload owners to correctly describe their api's so that our sdk generation can implement the correct functionality. In this case, not only is the description wrong, but so is this [documentation](https://learn.microsoft.com/en-us/graph/onenote-update-page#multipart-request-with-binary-content). The path 'v1.0/me/onenote/notebooks/pages/{page-id}/content' does not exist, more specifically there is not a navigation property in our metadata that would allow for "notebooks/pages/{pageId}", thus the documentation is incorrect. 
Furthermore, as you mentioned, the correct scehma should be /content not /microsoft.graph.onenotePatchContent. OneNotePatchContent is generated in our sdk because there is an action described in our metadata named 'OneNotePatchContent', however there is no documentation/indication that this action exists or is intended by OneNote. The parameters held by OnenotePageOnenotePatchContentParameterSet() are the correct parameters that we would expect in a request body, but the implementation is incorrect due to the misalignment that I have described above. We will look into contacting the workload owner for the oneNote api to get some further clarity for this documentation/api description issue.

**Comment by parkerschroeder** [comment]
Thanks for taking a look!

For anyone looking for a solution in the meantime, I helped work on a sample app that syncs notes to OneNote, which uses some modified classes from the graphsdk to update notes and co-wrote a guide on how to use it.

Sample app: https://github.com/microsoft/android-samples
Guide: https://devblogs.microsoft.com/surface-duo/microsoft-graph-android-3/

**Comment by baywet** [comment]
Hi,
Sorry for the late reply, we've been working hard [to release a new major version of the SDK](https://devblogs.microsoft.com/microsoft365dev/microsoft-graph-java-sdk-v6-now-generally-available-helps-you-write-simpler-faster-and-more-accurate-code/)
Since this is an issue with the older version, we'll be closing it for now.
Please go ahead and try with this new version, if you're still facing issues, feel free to open a new issue.
